### PR TITLE
Centrosymmetry

### PR DESCRIPTION
**Pull request by samwaseda**
Now via

```
neigh = structure.get_neighbors()
neigh.centrosymmetry
```

it should be possible to get the centro-symmetry value for the given environment.

Tests and DosString will follow.

### CODE REVIEW

**Comment by liamhuber** [comment]
What is the difference compared to `structure.analyze.pyscal_centro_symmetry`? Is it just to avoid writing `num_neighbors` a second time, or does this some how play well with the ragged version of `Neighbors` so that the CSP uses a different number of neighbors for each atom?

**Comment by samwaseda** [comment]
> What is the difference compared to `structure.analyze.pyscal_centro_symmetry`? Is it just to avoid writing `num_neighbors` a second time, or does this some how play well with the ragged version of `Neighbors` so that the CSP uses a different number of neighbors for each atom?

It's attached to `Neighbors`, so you can calculate the value for any point in the box, i.e. something like this is also possible:

```
neigh = structure.get_neighborhood(my_favourite_positions)
print(neigh.centrosymmetry)
```

File: pyiron_atomistics/atomistics/structure/neighbors.py [hunk 1]
```diff
@@ -584,6 +585,36 @@ def get_steinhardt_parameter(self, l, cutoff_radius=np.inf):
             for m in np.arange(-l, l+1)
         ], axis=0))
 
+    @staticmethod
+    def _get_all_possible_pairs(l):
```
Comment (samwaseda) [inline pyiron_atomistics/atomistics/structure/neighbors.py:589]: This function works maybe up to `l=10` but after that it takes too much memory and fails. If there's someone who can come up with a nice idea let me know. It in principle does what's described on [this page](https://stackoverflow.com/questions/5360220/how-to-split-a-list-into-pairs-in-all-possible-ways).

**Comment by samwaseda** [comment]
Following the pyiron meeting last week, we might change the internal algorithm, but since for the user the output is not going to change, I merge it now.

**Pull request merged**